rats-apps/src/python/rats/apps/_annotations.py

```diff
@@ -84,57 +88,91 @@ def get_namespaced_group(
             yield from c.get_namespaced_group(namespace, group_id)
 
 
+P = ParamSpec("P")
+
+
 def service(
-    service_id: ServiceId[T_ServiceType],
-) -> Callable[..., Callable[..., T_ServiceType]]:
+    service_id: ServiceId[T_ServiceType] | None = None,
```

Comment:
can we find a way to implement this without making it a fallback behavior? the meaning of the null value in this argument doesn't feel obvious to me and it feels like the author of providers makes a choice to define a specific id or generate one automatically. i'm unsure about using the auto generated value because of the need to make a container part of the public api, but would be up for trying it if we can decouple these two approaches

Reply:
So you would prefer keeping `service` as is and adding a new decorator like `autonamed_service`?

Reply:
yeah what ever name makes sense for you. i think we should be careful adding new APIs and should try to use new package names but generally I probably prefer adding a new function to modifying existing APIs for new behaviors

Reply:
Reverted the changes to this decorator and added `autoid_service` decorator instead.
Note that the syntax for using the new decorator is

```
    @autoid_service
    def my_service_method(self) -> ServiceType:
        ...
```

rather than

```
    @autoid_service()
    def my_service_method(self) -> ServiceType:
        ...
```

Are we happy with that?